src/ComposerRequireChecker/ASTLocator/LocateASTFromFiles.php

```diff
@@ -39,11 +45,21 @@ public function __invoke(Traversable $files): Traversable
                 continue;
             }
 
-            try {
-                $stmts = $this->parser->parse(file_get_contents($file), $this->errorHandler);
-            } catch (Error $e) {
-                // Convert the parse error into one which has the file information
-                throw new FileParseFailed($file, $e);
+            $key = sha1_file($file);
+            if (! $this->cache->has($key)) {
+                try {
+                    $stmts = $this->parser->parse(file_get_contents($file), $this->errorHandler);
+                } catch (Error $e) {
+                    // Convert the parse error into one which has the file information
+                    throw new FileParseFailed($file, $e);
+                }
+
+                $this->cache->set($key, $stmts);
```

Comment:
This caches the parsed AST, not the found symbols (which is what this tool actually processes).

Because this cache will then contain third-party symbols (defined by `nikic/php-parser`), I think this approach is not applicable.

Ideally, we should parse the resulting found symbols, which is a data structure "owned" by this package.

Reply:
It took some refactoring to be able to cache the symbols while only calling the parser on a cache miss. What do you think?